docs/installing-solidity.rst

```diff
@@ -321,8 +321,8 @@ The following are dependencies for all builds of Solidity:
 +===================================+=======================================================+
 | `CMake`_ (version 3.13+)          | Cross-platform build file generator.                  |
```

Comment:
Shouldn't this change to CMake 3.21.3 for 1.77?

Reply:
When you have a new Boost release, this information is added on the next release on CMake. So it's always the CMake version right after the Boost release date.

Reply:
Or at least it used to be this way.

Reply:
Haven't seen anything in CMake changelogs that references boost, and am not really sure why a boost upgrade would require a CMake bump as well (unless the boost directory structure or something similar changes). `3.13` is quite old, so perhaps we could bump that too though. @cameel @ekpyron Also @r0qs what version of CMake do we use on our ubu2204 images?

Reply:
It at the very least used to be true - i.e. the cmake files for finding boost were gloriously shipped with cmake itself not with boost, so if your boost was newer than your cmake, things broke...
IIRC they wanted to fix that, but not sure that's done just yet.

Reply:
I mentioned it before, but as a rule of thumb, I wouldn't bump minimum requirements beyond what current debian stable has without good reason - that'd be cmake 3.25.1...

Just to be clear: we're currently dropping support for building on vanilla ubuntu 20.04? What exactly would it take to be able to gracefully build there *and* on 22.04? If that was possible, I'd prefer it, but not if we have to get out of our way for it (I didn't follow the discussion last week precisely, since I was actually off when you bumped this).

Reply:
If we drop support for building on 20.04, we have to additionally drop ``focal`` from ``scripts/release_ppa.sh`` and add it to the list of versions that we copy the static binaries to in our release checklist.

Reply:
Yeah, I also think we should keep the requirement low enough to keep it working on distros people still use. I think Ubuntu 20.04 is still reasonable.

By the way, I actually wanted to say that we still have buildpack images for even older Ubuntu so it should be working but apparently not. The ossfuzz image is named after 16.04 but looking at the source, [it seems to be actually using 20.04](https://github.com/google/oss-fuzz/blob/master/infra/base-images/base-image/Dockerfile#L19). Should we rename it (not in this PR of course)?

Reply:
But yeah, IIRC now, the issue was that osx/homebrew bumped the boost version, causing builds to fail, and supporting both older and newer boost was tricky, so we bumped the minimum version, effectively dropping ubuntu 20.04 support, right? That'd be fine, but yeah, that'd mean we have to do https://github.com/ethereum/solidity/pull/13874#discussion_r1071525543, otherwise our release will fail.

Reply:
Right. Fine with dropping it then.

Reply:
Haha, yeah, I was wondering about the fuzzing image as well - we should probably just drop the ubuntu version from it altogether.

Reply:
But yeah, since the only backwards-incompatible change in https://github.com/ethereum/solidity/pull/13866 seems to be the lsp change, which we can drop eventually anyways, I wonder if we should just ``#ifdef`` that after all and could keep supporting 20.04 that way (would be enough to confirm build and tests locally without a full-blown CI run)...

Reply:
I.e. there's no call for dropping 20.04 in favour of a single ``#ifdef`` in code we're likely to drop at the next opportunity anyways :-).

Reply:
Going back to my original question, I _really_ think the CMake version on the table would be incompatible if you make the Boost 1.77 min version. it needs to be any CMake version released after Boost 1.77.